Incompatible types with types/react v18

This issue is a bug report.

## Steps to reproduce behavior

1. Install `@types/react@18.0.0`
2. Write some code importing this library:

```tsx
import React, { type ReactNode } from "react";
import BasicZoom from "react-medium-image-zoom";
import "react-medium-image-zoom/dist/styles.css";

export default function Zoom({ children }: { children: ReactNode }): JSX.Element {
  return <BasicZoom>{children}</BasicZoom>;
}
```

3. Run `yarn tsc`

## Expected behavior

This should compile

## Actual behavior

```
../node_modules/react-medium-image-zoom/dist/cjs/Controlled.d.ts(2,43): error TS2305: Module '"react"' has no exported member 'ReactType'.
```

The problem is that `ReactType` is removed in v18 in favor of `ElementType`. See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210

We can easily work around this by adding `"skipLibCheck": true`, but just would like you to know that such an issue exists. Moreover I noticed this lib is still using `@types/react@16` and `react@16`—any plans to upgrade? I can help with the process.

## Link to repository/example of behavior

https://github.com/facebook/docusaurus/runs/5952102932?check_suite_focus=true

--- a/source/Controlled.tsx
+++ b/source/Controlled.tsx
@@ -3,7 +3,7 @@ import React, {
   CSSProperties,
   FC,
   ReactNode,
-  ReactType,
+  ElementType,
   RefObject,
   StrictMode,
   memo,
@@ -25,7 +25,7 @@ export interface ControlledProps {
   portalEl?: HTMLElement
   scrollableEl?: HTMLElement | Window
   transitionDuration?: number
-  wrapElement?: ReactType
+  wrapElement?: ElementType
   wrapStyle?: CSSProperties
   zoomMargin?: number
   zoomZindex?: number

--- a/source/Uncontrolled.tsx
+++ b/source/Uncontrolled.tsx
@@ -3,7 +3,7 @@ import React, {
   CSSProperties,
   FC,
   ReactNode,
-  ReactType,
+  ElementType,
   RefObject,
   StrictMode,
   memo,
@@ -23,7 +23,7 @@ export interface UncontrolledProps {
   portalEl?: HTMLElement
   scrollableEl?: HTMLElement | Window
   transitionDuration?: number
-  wrapElement?: ReactType
+  wrapElement?: ElementType
   wrapStyle?: CSSProperties
   zoomMargin?: number
   zoomZindex?: number

